src/main/scala/org/broadinstitute/dsde/workbench/sam/model/api/SamUserAllowances.scala

```diff
@@ -2,19 +2,26 @@ package org.broadinstitute.dsde.workbench.sam.model.api
 
 import spray.json.DefaultJsonProtocol._
 import spray.json.RootJsonFormat
-
 object SamUserAllowances {
-  implicit val SamUserAllowedResponseFormat: RootJsonFormat[SamUserAllowances] = jsonFormat2(SamUserAllowances.apply)
+  implicit val SamUserAllowedResponseFormat: RootJsonFormat[SamUserAllowances] = jsonFormat2(SamUserAllowances.apply(_: Boolean, _: SamUserAllowancesDetails))
 
-  def apply(allowed: Boolean, enabled: Boolean, termsOfService: Boolean): SamUserAllowances =
-    SamUserAllowances(allowed, Map("enabled" -> enabled, "termsOfService" -> termsOfService))
+  def apply(enabled: Boolean, termsOfService: Boolean): SamUserAllowances =
+    SamUserAllowances(enabled && termsOfService, SamUserAllowancesDetails(enabled, termsOfService))
 }
 final case class SamUserAllowances(
     allowed: Boolean,
-    details: Map[String, Boolean]
+    details: SamUserAllowancesDetails
```

Comment:
Why is `allowed` not needed any more? Should it be removed from here too?

Reply:
`allowed` is a convenience parameter that just ANDs the different things that could deny a user access. Really useful for consumers, as they don't have to parse the details to know whether or not a user can use Terra